assignments/prototypes.js

```diff
@@ -7,22 +7,39 @@
   
   Each constructor function has unique properties and methods that are defined in their block comments below:
 */
-  
+
 /*
   === GameObject ===
   * createdAt
   * name
   * dimensions (These represent the character's size in the video game)
   * destroy() // prototype method that returns: `${this.name} was removed from the game.`
 */
-
+class GameObject {
+  constructor(attr) {
+    this.createdAt = attr.createdAt;
+    this.name = attr.name;
+    this.dimensions = attr.dimensions;
+  }
+  destroy() {
+    return `${this.name} was removed from the game. `;
+  }
+}
 /*
   === CharacterStats ===
   * healthPoints
   * takeDamage() // prototype method -> returns the string '<object name> took damage.'
   * should inherit destroy() from GameObject's prototype
 */
-
+class CharacterStats extends GameObject {
+  constructor(attr) {
+    super(attr);
+    this.healthPoints = attr.healthPoints;
+  }
+  takeDamage() {
+    return `${this.name} took damage`;
```

Comment:
Although it doesn't suggest it in the directions, you could pass in an argument for the actual 'dmg' and print that argument in the return statement.
``` 
takeDamage(dmg) {
    return `${this.name} took ${dmg} damage`;
```